l2geth/rollup/sync_service.go

```diff
@@ -1393,6 +1445,33 @@ func (s *SyncService) syncTransactionBatchRange(start, end uint64) error {
 	return nil
 }
 
+func (s *SyncService) syncEigenTransactionBatchRange(start, end uint64) error {
+	log.Info("Syncing eigen transaction batch range", "start", start, "end", end)
+	for i := start; i <= end; i++ {
+		log.Debug("Fetching transaction batch", "index", i)
+		rollupInfo, err := s.eigenClient.GetRollupStoreByRollupBatchIndex(int64(i))
+		if err != nil {
+			return fmt.Errorf("cannot get rollup store by batch index: %w", err)
+		}
+		txs, err := s.eigenClient.GetBatchTransactionByDataStoreId(rollupInfo.DataStoreId)
+		if err != nil {
+			return fmt.Errorf("cannot get eigen transaction batch: %w", err)
+		}
+		for _, tx := range txs {
+			if tx.QueueOrigin().String() != "sequencer" || tx.GetMeta().Index == nil {
```

Comment:
Why skip L1ToL2 transactions? This type of transaction is found in Indexed Txs.
 If a transaction without Index is synchronized, the data source is not reliable,Whether to terminate the process？